src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -361,8 +466,14 @@ private BoundExpression MakeEqual(BoundExpression loweredLiteral, BoundExpressio
                 case SpecialType.System_UInt64:
                     return MakeBinaryOperator(_factory.Syntax, BinaryOperatorKind.ULongEqual, loweredLiteral, input, booleanType, method: null);
                 default:
-                    // PROTOTYPE(patterns2): need more efficient code for enum test, e.g. `color is Color.Red`
-                    // This is the (correct but inefficient) fallback for any type that isn't yet implemented (e.g. enums)
+                    if (loweredLiteral.Type.IsEnumType())
+                    {
+                        return MakeBinaryOperator(_factory.Syntax, BinaryOperatorKind.EnumEqual, loweredLiteral, input, booleanType, method: null);
+                    }
+
+                    // This is the (correct but inefficient) fallback for any type that isn't yet implemented.
+                    // However, the above should handle all types.
+                    Debug.Assert(false); // don't fail in non-debug builds
```

Comment:
>Debug.Assert(false); // don't fail in non-debug builds [](start = 20, length = 54)

Should we `throw ExceptionUtilities.UnexpectedValue(loweredLiteral.Type);` instead of generating code in an untested codepath?

Reply:
The code actually was tested until I added support for decimal.